isvcs/zk.go

```diff
@@ -139,27 +154,49 @@ func SetZKHealthCheck(connectString string) HealthCheckFunction {
 // This function sets up a ZooKeeper quorum check function parameterized with the connection string eg: "127.0.0.1:2181".
 func SetZKQuorumCheck(connectString string) HealthCheckFunction {
 	return func(halt <-chan struct{}) error {
+		healthy := true
+		times := -1
 
 		for {
+			if !healthy && times == 3 {
+				log.WithFields(logrus.Fields{
```

Comment:
Ditto previous comment about reducing repetitive `WithFields` calls in all of the log messages in this method.